webui/react/src/pages/Settings/UserManagement.test.tsx

```diff
@@ -61,8 +57,8 @@ const Container: React.FC = () => {
   const [canceler] = useState(new AbortController());
   const fetchUsers = useFetchUsers(canceler);
 
-  const loadUsers = useCallback(async () => {
-    await fetchUsers();
+  const loadUsers = useCallback(() => {
+    fetchUsers();
```

Comment:
**question:** maybe the original code here is right but that the `fetchUsers` is defined improperly? Shouldn't [line 63 of `useFetchUsers`](https://github.com/determined-ai/determined/blob/master/webui/react/src/stores/users.tsx#L63) be this instead?
```
export const useFetchUsers = (canceler: AbortController): ((cfg?: FetchUsersConfig) => Promise<void>) => {
```
Note the `Promise<void>` instead of `void` for the callback return

Reply:
So tests are a different beast but generally **actions** should not be awaitable in reactive programming, hence the `void`.

Reply:
but the [line 63 of useFetchUsers](https://github.com/determined-ai/determined/blob/master/webui/react/src/stores/users.tsx#L63) caleb mentioned actually returns `Promise<void>` (i didnt notice it).
is that still ok to be `void`?

Reply:
I see. The test can still be written without the `async/await` but shouldn't the returned typescript definition of the resulting callback still just send back `Promise<void>` because that is what is actually being returned?

Reply:
The idea here is that the function _should not_ be returning the Promise. Returning the promise allows users to synchronously await state updates instead of using `useEffect` which is the correct approach since it moves the dependency tree into the reactive system.

Reply:
Just to make this concrete: imagine two actions with a one-way dependency. If they return promises you might be tempted to do this:
```typescript
useEffect(async () => {
   await fetchUser()
   fetchUserRoles(user)
}, [user])
```
When the correct code is actually this:
```typescript
useEffect(() => fetchUser(), [])
useEffect(() => fetchUserRoles(user), [user])
```
Now, the bug here is very obvious, but usually you run into much more subtle bugs. Essentially what you're doing is maintaining two separate dependency graphs that don't know about each other which adds a lot of hidden complexity.